app/controllers/repos_controller.rb

```diff
@@ -24,10 +24,16 @@ def show
     if (@form_result_type = params[:result_type]) &&
        (@benchmark_type = find_benchmark_type_by_category(@form_result_type))
 
+      # read versions from cache since it's shared among all `@charts`
+      version_cache_key = "charts:#{@benchmark_type.id}:#{@benchmark_run_display_count}"
+      versions = $redis.get(version_cache_key)
+      @versions = JSON.parse(versions) if versions
+
+      versions_calculate_once = ActiveSupport::OrderedHash.new
       @charts = @benchmark_type.benchmark_result_types.map do |benchmark_result_type|
         cache_key = "#{BenchmarkRun.charts_cache_key(@benchmark_type, benchmark_result_type)}:#{@benchmark_run_display_count}"
 
-        if (columns = $redis.get(cache_key))
+        if (columns = $redis.get(cache_key)) && (versions)
```

Comment:
`&&` has higher precedence than `=` so we need the brackets around `columns = ...`, we don't need the brackets around `versions` though